pilot/pkg/features/pilot.go

```diff
@@ -638,10 +638,6 @@ var (
 		"If enabled, pilot will only send the delta configs as opposed to the state of the world on a "+
 			"Resource Request. This feature uses the delta xds api, but does not currently send the actual deltas.").Get()
 
-	MetadataDiscovery = env.Register("ISTIO_METADATA_DISCOVERY",
-		false,
-		"Enables proxy discovery of the workload metadata to back-fill the telemetry reports.").Get()
```

Comment:
Yeah, that was just not completed for sidecars. It's auto-enabled for waypoints and not enabled at all for sidecars. Do you want me to finish it?

Reply:
Reading the name or the comment - I can't figure out what it does.DISCOVERY is a super generic term.

Is this the 'Use WDS instead of metadata exchange' ? If it is, I think leaving it default for ambient - and not implementing it for now in sidecars is reasonable. But I would add it to Gateways too.

If it's not a lot of effort - add it to sidecars too.  I don't think it needs a separate setting - it can be conditional on
ambient controller enabled. Users who don't enable ambient - won't have it anyways, and if they do enable
ambient - it'll be likely useful.

Reply:
> Is this the 'Use WDS instead of metadata exchange' ? If it is, I think leaving it default for ambient - and not implementing it for now in sidecars is reasonable. But I would add it to Gateways too.

Yes, that's the state it is in - implemented and enabled in waypoints, but not in sidecars. 

> If it's not a lot of effort - add it to sidecars too. I don't think it needs a separate setting - it can be conditional on ambient controller enabled. Users who don't enable ambient - won't have it anyways, and if they do enable ambient - it'll be likely useful.

I think we can wait for one more release - there's no urgency to add it to sidecars. If WDS changes, we're in worse situation with sidecars. The service is implemented by istiod, there's no separate controller AFAIK.